quic/s2n-quic-transport/src/endpoint/mod.rs

```diff
@@ -1010,8 +1026,10 @@ impl<Cfg: Config> Endpoint<Cfg> {
             .create_client_peer_id_registry(internal_connection_id, rotate_handshake_connection_id);
 
         let congestion_controller = {
-            let path_info =
-                congestion_controller::PathInfo::new(self.mtu_config.initial_mtu, &remote_address);
+            let path_info = congestion_controller::PathInfo::new(
+                self.endpoint_mtu_config.initial_mtu(),
```

Comment:
shouldn't the congestion controller get the actual mtu config being used for the connection?

Reply:
definitely

Reply:
Yes good catch. I changed the signature of `PathInfo::new` to take CheckedConfig.

This was one of the reason to have a new type. I wanted to enforce that we accidentally didn't pass a non-checked one (either endpoint or connection).

The other reason was any pub method we add to impl Config also become part of the pub API (it needs to be pub so that transport crate can call this method). `fn checked(info: ConnectionInfo, endpoint_config: Config) -> Config` is not something the Application should be calling.

Reply:
It's a little weird though, since the mtu config from the endpoint _is_ checked, maybe that one should be a different type? 

What if you had another trait, something like the `Validator` trait we have for the connection ID provider. And that had a method `get_checked(endpoing_config: EndpointConfig) -> mtu::Config`. And we implement that for anything that implements the `Configurator` trait. The mtu provider would then return Validator (or some other better name), so we know we're only getting checked mtu configs out of it